code/modules/clothing/spacesuits/rig/rig_attackby.dm

```diff
@@ -153,53 +153,63 @@
 							cell.forceMove(user.put_in_hands(cell))
 						cell = null
 					else
-						to_chat(user, "There is nothing loaded in that mount.")
+						to_chat(user, SPAN_WARNING("There is nothing loaded in that mount."))
 
 				if("system module")
-
 					var/list/possible_removals = list()
 					for(var/obj/item/rig_module/module in installed_modules)
 						if(module.permanent)
 							continue
 						possible_removals[module.name] = module
 
 					if(!possible_removals.len)
-						to_chat(user, "There are no installed modules to remove.")
+						to_chat(user, SPAN_WARNING("There are no installed modules to remove."))
 						return
 
 					var/removal_choice = input("Which module would you like to remove?") as null|anything in possible_removals
 					if(!removal_choice)
 						return
 
 					var/obj/item/rig_module/removed = possible_removals[removal_choice]
-					to_chat(user, "You detatch \the [removed] from \the [src].")
+					to_chat(user, SPAN_NOTICE("You detach \the [removed] from \the [src]."))
 					removed.forceMove(get_turf(src))
 					removed.removed()
 					installed_modules -= removed
 					update_icon()
 
-		else if(istype(W,/obj/item/stack/nanopaste)) //EMP repair
+		else if(istype(W, /obj/item/stack/nanopaste)) //EMP repair
 			var/obj/item/stack/S = W
 			if(malfunctioning || malfunction_delay)
 				if(S.use(1))
-					to_chat(user, "You pour some of \the [S] over \the [src]'s control circuitry.")
+					to_chat(user, SPAN_NOTICE("You pour some of \the [S] over \the [src]'s control circuitry."))
 					malfunctioning = 0
 					malfunction_delay = 0
 				else
-					to_chat(user, "\The [S] is empty!")
-
+					to_chat(user, SPAN_WARNING("\The [S] is empty!"))
 		return
 
 	// If we've gotten this far, all we have left to do before we pass off to root procs
 	// is check if any of the loaded modules want to use the item we've been given.
 	for(var/obj/item/rig_module/module in installed_modules)
-		if(module.accepts_item(W,user)) //Item is handled in this proc
+		if(module.accepts_item(W, user)) //Item is handled in this proc
 			return
-	..()
 
+	if(!open)
```

Comment:
this shouldn't work if it's on your bag

Reply:
this is also a bit too risk-free and too fast for my tastes, should be a do_after with a chance of things going wrong tbh